source/common/chromium_url/url_canon.h

```diff
@@ -146,7 +146,8 @@ template <typename T, int fixed_capacity = 1024> class RawCanonOutputT : public
 
   void Resize(int sz) override {
     T* new_buf = new T[sz];
-    memcpy(new_buf, this->buffer_, sizeof(T) * (this->cur_len_ < sz ? this->cur_len_ : sz));
+    memcpy(new_buf, this->buffer_, // NOLINT(safe-memcpy)
```

Comment:
I think in this case it might make sense to use MemBlockBuilder in lieu of the call `new`, e.g.
```
MemBlockBuilder<T> new_buf(sz);
new_buf.appendData(absl::Span<T>(this->buffer, std::min(this->cur_len_, sz));
```
and then `this->buffer_ = new_buf.release().release();`

Reply:
Actually, another method was added, ```releasePointer()```, which returns ```owned_data_.release()```, unless it isn’t set when constructing MemBlockBuilder. This application seems very appropiate and, although in the next push I will probably only add those changes suggested, I will use it as an example to move forward in this and future PRs (if the idea is still to keep this one separeted from the migration of MemBlockBuilder into BufferImpl).